lib/url/URLParser.js

```diff
@@ -0,0 +1,37 @@
+/*
+	MIT License http://www.opensource.org/licenses/mit-license.php
+	Author Tobias Koppers @sokra
+*/
+
+"use strict";
+
+const path = require("path");
+const { RawSource } = require("webpack-sources");
+
+class URLParser {
+	constructor(options = {}) {
+		this.options = options;
+	}
+
+	parse(source, state) {
+		const { module } = state;
+
+		if (typeof source === "string") {
```

Comment:
I think this parser can be empty, since there is nothing to parse in a raw asset (similar to the WASM parser)

Reply:
Depending on filetype, some file may have dependencies. I. e. svg can reference other files. These dependencies needed to be parsed here. Maybe we can separate module types for these assets: `url/svg`. Not sure if there are more.

It also need a special generator to replace url references in the source.

Actually this is pretty similar to HTML, so we might skip it for now and take care of this in the HTML PR

Reply:
Good catch :) Setting `NormalModule.binary` already handles that